sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.aest.reissueMSFAA.e2e-spec.ts

```diff
@@ -0,0 +1,355 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import * as request from "supertest";
+import {
+  AESTGroups,
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  getAESTToken,
+} from "../../../testHelpers";
+import {
+  E2EDataSources,
+  MSFAAStates,
+  createE2EDataSources,
+  createFakeMSFAANumber,
+  saveFakeApplication,
+  saveFakeApplicationDisbursements,
+  saveFakeStudent,
+} from "@sims/test-utils";
+import {
+  ApplicationStatus,
+  DisbursementScheduleStatus,
+  OfferingIntensity,
+} from "@sims/sims-db";
+
+describe("ApplicationAESTController(e2e)-reissueMSFAA", () => {
+  let app: INestApplication;
+  let db: E2EDataSources;
+
+  beforeAll(async () => {
+    const { nestApplication, dataSource } = await createTestingAppModule();
+    app = nestApplication;
+    db = createE2EDataSources(dataSource);
+  });
+
+  it("Should reissue an MSFAA and associate with both disbursements when both disbursements are pending and the current MSFAA is signed but canceled.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    const currentMSFAA = createFakeMSFAANumber(
+      { student },
+      {
+        state: MSFAAStates.Signed | MSFAAStates.CancelledOtherProvince,
```

Comment:
won't this give the sum of 1 and 4 that is 5 
![image](https://github.com/bcgov/SIMS/assets/77353155/f0cab7d4-3b24-482f-9640-299bec3bbe63)
does, it satisfy the condition that we need?

Reply:
Yes, please review how the check is done.

Reply:
Sorry but I didn't understand why using the bitwise operations here.

Reply:
@andrepestana-aot please let me know if need more information and please review the code for the `createFakeMSFAANumber` method.
https://github.com/bcgov/SIMS/pull/1928#discussion_r1189112182